lib/shopify_cli/command.rb

```diff
@@ -31,6 +31,12 @@ def call(args, command_name, *)
         end
       rescue OptionParser::InvalidOption => error
         arg = error.args.first
+        store_name = arg.match(/\A--(?<store_name>.*\.myshopify\.com)\z/)&.[](:store_name)
+        if store_name && !arg.match?(/\A--(store|shop)=/)
```

Comment:
Sweet! What about adding a unit test to prevent regressions in this logic in the future? If having this logic as part of the command makes it less unit-testable, we can extract it elsewhere.